src/providers/supabase/authProvider.ts

```diff
@@ -20,20 +21,12 @@ const baseAuthProvider = supabaseAuthProvider(supabase, {
             avatar: data.avatar?.src,
         };
     },
-    getPermissions: async user => {
-        const { data, error } = await supabase
-            .from('sales')
-            .select('administrator')
-            .match({ user_id: user.id })
-            .single();
-
-        if (!data || error) {
-            return null;
-        }
-
-        return data?.administrator ? 'admin' : 'user';
-    },
 });
+// FIXME: Now that react-admin pessimistically calls getPermissions, it calls getPermissions when it initializes its routes
```

Comment:
Actually this is because of the default implementation of `getPermissions` in `ra-supabase`, which throws when the user is not authenticated.
Should be fixed by https://github.com/marmelab/ra-supabase/pull/79.

Reply:
That's indeed a bug, but not a regression : it comes from ra-supabase. We will be able to remove this code once https://github.com/marmelab/ra-supabase/pull/79 is merged and published.